From 8bd9ba6a497ae4b8594bebb0c7242a00d96ee386 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 23 Jan 2026 14:18:14 +0100
Subject: [PATCH] units/getty@.service: use [Install]Alias= instead of static
 alias

In Fedora, kmsconvt@.service is starting to be used instead of getty@.service
to have nicer font handling. This means that we need the autovt@.service alias
point to the new unit. So far the alias was done through a static symlink
because there was little reason to change it. Let's use [Install] instead so
the decision which implementation to use can be made after installation.

https://bugzilla.redhat.com/show_bug.cgi?id=2429626
https://fedoraproject.org/wiki/Changes/UseKmsconVTConsole
---
 units/getty@.service.in | 3 +++
 units/meson.build       | 1 -
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/units/getty@.service.in b/units/getty@.service.in
index 19b95ce56d45c..e7e2aab045fd9 100644
--- a/units/getty@.service.in
+++ b/units/getty@.service.in
@@ -47,6 +47,7 @@ KillMode=process
 {% endif %}
 IgnoreSIGPIPE=no
 SendSIGHUP=yes
+
 ImportCredential=tty.virtual.%I.agetty.*:agetty.
 ImportCredential=tty.virtual.%I.login.*:login.
 ImportCredential=agetty.*
@@ -58,5 +59,7 @@ ImportCredential=shell.*
 UnsetEnvironment=LANG LANGUAGE LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT LC_IDENTIFICATION
 
 [Install]
+Alias=autovt@.service
+
 WantedBy=getty.target
 DefaultInstance=tty1
diff --git a/units/meson.build b/units/meson.build
index f2c5d220554b5..36dcb3f117483 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -45,7 +45,6 @@ units = [
         },
         {
           'file' : 'getty@.service.in',
-          'symlinks' : ['autovt@.service'],
         },
         {
           'file' : 'graphical.target',
